src/main/scala/org/ergoplatform/nodeView/wallet/persistence/OffChainRegistry.scala

```diff
@@ -0,0 +1,67 @@
+package org.ergoplatform.nodeView.wallet.persistence
+
+import org.ergoplatform.nodeView.history.ErgoHistory
+import org.ergoplatform.wallet.boxes.TrackedBox
+
+/**
+  * Holds version-agnostic indexes (such as off-chain boxes) in runtime memory.
+  */
+final case class OffChainRegistry(height: Int,
+                                  offChainBalances: Seq[Balance],
+                                  onChainBalances: Seq[Balance]) {
+
+  import org.ergoplatform.nodeView.wallet.IdUtils._
+
+  /**
+    * Off-chain index considering on-chain balances.
+    */
+  val readIndex: RegistryIndex = {
+    val balance = offChainBalances.map(_.value).sum + onChainBalances.map(_.value).sum
+    val tokensBalance = (offChainBalances ++ onChainBalances)
+      .flatMap(_.assets)
+      .foldLeft(Map.empty[EncodedTokenId, Long]) { case (acc, (id, amt)) =>
+        acc.updated(id, acc.getOrElse(id, 0L) + amt)
+      }
+    RegistryIndex(height, balance, tokensBalance, Seq.empty)
+  }
+
+  def updated(certainBoxes: Seq[TrackedBox],
```

Comment:
Please add method annotation, including parameters.
The same for `updateOnBlock`, it is not clear what is the difference between them and what is the difference between `certainBoxes` and `allCertainBoxes` in these methods